test/e2e/cassandracluster_test.go

```diff
@@ -277,6 +279,159 @@ func cassandraClusterUpdateConfigMapTest(t *testing.T, f *framework.Framework, c
 	assert.Equal(t, cluster.Spec.ConfigMapName, updatedCluster.Spec.ConfigMapName)
 }
 
+func cassandraClusterCleanupTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) {
+	namespace, _ := ctx.GetNamespace()
+	clusterName := "cassandra-e2e"
+
+	logrus.Debugf("Creating cluster")
+
+	cluster := mye2eutil.HelperInitCluster(t, f, ctx, "cassandracluster-1DC-no-autopilot.yaml", namespace)
+
+	if err := f.Client.Create(goctx.TODO(), cluster, &framework.CleanupOptions{}); err != nil && !apierrors.IsAlreadyExists(err) {
+		t.Fatalf("Error Creating CassandraCluster: %v", err)
+	}
+
+	waitForClusterToBeReady(cluster, f, t)
+
+	cluster = getCassandraCluster(clusterName, namespace, f, t)
+
+	selector := metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      "statefulset.kubernetes.io/pod-name",
+				Operator: metav1.LabelSelectorOpIn,
+				Values:   []string{"cassandra-e2e-dc1-rack1-0", "cassandra-e2e-dc1-rack2-0"},
+			},
+		},
+	}
+	opts := metav1.ListOptions{
+		LabelSelector: metav1.FormatLabelSelector(&selector),
+	}
+
+	pods := &v1.PodList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+	}
+
+	pods, err := f.KubeClient.CoreV1().Pods(namespace).List(opts)
+	if err != nil {
+		t.Fatalf("Failed to list pods: %s", err)
+	}
+	assert.Equal(t, 2, len(pods.Items))
+
+	for _, pod := range pods.Items {
+		pod.Labels["operation-name"] = "cleanup"
+		pod.Labels["operation-status"] = "ToDo"
+		if err := f.Client.Update(goctx.TODO(), &pod); err != nil {
+			t.Fatalf("pod update failed: %s", err)
+		}
+	}
+
+	checkCleanupExecuted := func(rack string, node int, cc *api.CassandraCluster) (bool, error) {
+		dcRack := "dc1-" + rack
+		nodeName := fmt.Sprintf("%s-%s-%d", clusterName, dcRack, node)
+
+		//dcRackStatus, found :=cc.Status.CassandraRackStatus[dcRack]
+		_, found :=cc.Status.CassandraRackStatus[dcRack]
+		if !found {
+			return false, fmt.Errorf("Did not find rack status for %s", rack)
+		}
+
+		// Commenting out status checks for now due to https://github.com/Orange-OpenSource/cassandra-k8s-operator/issues/98
+		//
+		//if len(dcRackStatus.PodLastOperation.PodsOK) == 0 {
```

Comment:
If that code is no longer relevant, let's get rid of it

Reply:
I rebased on master and added back these checks.